Netlogo 6.2.0 breaks gis:import-wms-drawing

On NetLogo 6.1.1 with GIS extension 1.2.0, this code runs perfectly fine
```NetLogo
extensions [ gis ]
to setup
  clear-all
  gis:set-world-envelope (list -180 180 -90 90)
end
to draw-background-image
  gis:import-wms-drawing "https://ows.terrestris.de/osm/service?" "EPSG:4326" "OSM-WMS" 128
end
```

On 6.2.0 with GIS extension 1.2.0, it throws this exception
```
Extension exception: com/sun/image/codec/jpeg/ImageFormatException
error while observer running GIS:IMPORT-WMS-DRAWING
  called by procedure DRAW-BACKGROUND-IMAGE
  called by Button 'draw-background-image'
org.nlogo.nvm.WrappedExtensionException: Extension exception: com/sun/image/codec/jpeg/ImageFormatException
 at org.nlogo.prim._extern.perform(_extern.java:40)
 at org.nlogo.nvm.Context.stepConcurrent(Context.java:107)
 at org.nlogo.nvm.ConcurrentJob.step(ConcurrentJob.scala:65)
 at org.nlogo.job.JobThread.runPrimaryJobs(JobThread.scala:133)
 at org.nlogo.job.JobThread.$anonfun$run$1(JobThread.scala:68)
 at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
 at scala.util.control.Exception$Catch.apply(Exception.scala:228)
 at org.nlogo.api.Exceptions$.handling(Exceptions.scala:41)
 at org.nlogo.job.JobThread.run(JobThread.scala:66)
Caused by: org.nlogo.api.ExtensionException: com/sun/image/codec/jpeg/ImageFormatException
 at com.sun.media.jai.codecimpl.JPEGImageDecoder.decodeAsRenderedImage(JPEGImageDecoder.java:46)
 at com.sun.media.jai.codec.ImageDecoderImpl.decodeAsRenderedImage(ImageDecoderImpl.java:140)
 at org.myworldgis.netlogo.LoadWMSImage.performInternal(LoadWMSImage.java:86)
 at org.myworldgis.netlogo.GISExtension$Command.perform(GISExtension.java:61)
 at org.nlogo.prim._extern.perform(_extern.java:36)
 ... 8 more
NetLogo 6.2.0
main: org.nlogo.app.AppFrame
thread: JobThread
OpenJDK 64-Bit Server VM 1.8.0_275 (BellSoft; 1.8.0_275-b01)
operating system: Mac OS X 10.16 (x86_64 processor)
Scala version 2.12.12
JOGL: (3D View not initialized)
OpenGL Graphics: (3D View not initialized)
model: wms_regression
02:28:29.751 RuntimeErrorEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
02:28:28.796 AddJobEvent (org.nlogo.window.ButtonWidget) AWT-EventQueue-0
02:28:28.751 InputBoxLoseFocusEvent (org.nlogo.window.ButtonWidget) AWT-EventQueue-0
02:28:28.727 PeriodicUpdateEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
02:28:28.525 PeriodicUpdateEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
02:28:28.320 JobRemovedEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) JobThread
02:28:28.319 TickStateChangeEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) JobThread
02:28:28.304 PeriodicUpdateEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
02:28:28.253 OutputEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
02:28:28.248 TickStateChangeEvent (org.nlogo.app.App$$anon$4 (org.nlogo.window.GUIWorkspace)) JobThread
```

--- a/LICENSE.md
+++ b/LICENSE.md
@@ -7,9 +7,9 @@ The Java source code (files in the src directory whose names end in ".java") and
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
 
- * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
- * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
- * The names of its contributors may not be used to endorse or promote products derived from this software without specific prior written permission.
+* Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+* The names of its contributors may not be used to endorse or promote products derived from this software without specific prior written permission.
 
 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
@@ -69,9 +69,8 @@ ESRI is a trademark of Environmental Systems Research Institute, Inc.
 
 --------------------------------------------------------------------------------
 
-The Java Advanced Imaging libraries (the binary files jai_codec-1.1.3.jar and
-jai_core-1.1.3.jar) are copyright (c) 2006 Sun Microsystems, and are made 
-available under the following license:
+The Java Advanced Imaging libraries (the binary file jai_core-1.1.3.jar) are copyright
+(c) 2006 Sun Microsystems, and are made available under the following license:
 
 Sun Microsystems, Inc.
 Binary Code License Agreement

--- a/README.md
+++ b/README.md
@@ -1665,9 +1665,9 @@ The Java source code (files in the src directory whose names end in ".java") and
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
 
- * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
- * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
- * The names of its contributors may not be used to endorse or promote products derived from this software without specific prior written permission.
+* Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+* The names of its contributors may not be used to endorse or promote products derived from this software without specific prior written permission.
 
 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
@@ -1727,9 +1727,8 @@ ESRI is a trademark of Environmental Systems Research Institute, Inc.
 
 --------------------------------------------------------------------------------
 
-The Java Advanced Imaging libraries (the binary files jai_codec-1.1.3.jar and
-jai_core-1.1.3.jar) are copyright (c) 2006 Sun Microsystems, and are made 
-available under the following license:
+The Java Advanced Imaging libraries (the binary file jai_core-1.1.3.jar) are copyright
+(c) 2006 Sun Microsystems, and are made available under the following license:
 
 Sun Microsystems, Inc.
 Binary Code License Agreement

--- a/build.sbt
+++ b/build.sbt
@@ -21,7 +21,6 @@ libraryDependencies ++= Seq(
   "commons-codec"              % "commons-codec"      % "1.10",
   "org.ngs"                    % "ngunits"            % "1.0.0" from "https://s3.amazonaws.com/ccl-artifacts/ngunits-1.0.jar",
   "javax.media"                % "jai_core"           % "1.1.3" from "https://s3.amazonaws.com/ccl-artifacts/jai_core-1.1.3.jar",
-  "com.sun.media"              % "jai_codec"          % "1.1.3" from "https://s3.amazonaws.com/ccl-artifacts/jai_codec-1.1.3.jar", 
   "org.tinfour"                % "TinfourCore"        % "2.1.5",
   "com.googlecode.json-simple" % "json-simple"        % "1.1.1")
 

--- a/src/org/myworldgis/netlogo/LoadWMSImage.java
+++ b/src/org/myworldgis/netlogo/LoadWMSImage.java
@@ -4,16 +4,15 @@
 
 package org.myworldgis.netlogo;
 
-import com.sun.media.jai.codec.ImageCodec;
 import com.vividsolutions.jts.geom.Envelope;
 import java.awt.Dimension;
 import java.awt.Graphics2D;
 import java.awt.geom.AffineTransform;
 import java.awt.image.BufferedImage;
-import java.awt.image.RenderedImage;
 import java.awt.image.WritableRaster;
 import java.io.IOException;
 import java.io.PushbackInputStream;
+import javax.imageio.ImageIO;
 import org.apache.commons.httpclient.HttpStatus;
 import org.apache.commons.httpclient.methods.GetMethod;
 import org.myworldgis.util.HttpClientManager;
@@ -70,26 +69,16 @@ public class LoadWMSImage extends GISExtension.Command {
         Dimension pixelBounds = new Dimension(drawing.getWidth(), drawing.getHeight());
         String url = WMSUtils.makeGetMapURL(serverURL, viewBounds, pixelBounds, srs, layers, WMSUtils.ImageFormat.JPEG);
         GetMethod method = new GetMethod(url);
-        RenderedImage image = null;
+        BufferedImage image = null;
         try {
             int statusCode = HttpClientManager.getInstance().execute(method);
             if (statusCode != HttpStatus.SC_OK) {
-                throw(new IOException(HttpClientManager.errorMsg(statusCode, url)));
+                throw (new ExtensionException("Unable to fetch wms image:\n" + HttpClientManager.errorMsg(statusCode, url)));
             }
             PushbackInputStream in = new PushbackInputStream(method.getResponseBodyAsStream(), 12);
-            byte[] beginning = new byte[12];
-            int bytesRead = in.read(beginning, 0, 12);
-            in.unread(beginning, 0, bytesRead);
-            if (bytesRead < 12) {
-                throw new ExtensionException( org.nlogo.util.Utils.reader2String( new java.io.InputStreamReader( in ) ) ) ;
-            } else if ("JFIF".equals(new String(beginning, 6, 4))) {
-                image = ImageCodec.createImageDecoder("JPEG", in, null).decodeAsRenderedImage();
-            } else if ("PNG".equals(new String(beginning, 1, 3))) {
-                // sometimes you get a PNG even when you ask for a JPEG
-                image = ImageCodec.createImageDecoder("PNG", in, null).decodeAsRenderedImage();
-            } else {
-                throw new ExtensionException( org.nlogo.util.Utils.reader2String( new java.io.InputStreamReader( in ) ) ) ;
-            }
+            image = ImageIO.read(in);
+        } catch (IOException e) {
+            throw new ExtensionException("Unable to decode wms image downloaded from: " + url);
         } finally {
             method.releaseConnection();
         }

